### ISSUE TITLE
Home Screen Refresh

### ISSUE THREAD

**Issue by ajsarkar28** [department-of-veterans-affairs/va-mobile-app#8183]
AC:
- When a Veteran does an action that affects the feature values displayed on the home screen (read a secure message, cancel an appointment, fill a prescription that is available for refill), then the feature value's state is captured and reflected on the home screen
- API refresh timer is set to 5 min
- Use focus effect on the home screen is retired
- Maintain scroll position on backtrack to Homescreen from any of the links 
- Note that this ticket is blocked by the react query migration tickets

During dev/QA, should test that the loading behavior previously on the home screen when returning from "Contact VA" does not demonstrate the jumpiness in this [video] (https://drive.google.com/file/d/1fe415bCgN77R-LqtPtZfGyzse04t7RbW/view?usp=sharing).

**Comment by theodur** [comment]
Most of the work for this has been completed in the linked branch. The remainder of the work (refreshing the appointments and unread messages count) can't be finished until the React Query migration for those features are merged into develop.

**Comment by theodur** [comment]
Since it seems unlikely that the React Query secure messaging migration will be merged any time soon, I went ahead and implemented the refresh logic for secure messages in Redux. I didn't do the same for appointments because the appointments React Query migration PR will likely be merged soon, and we'd need to update it to use the new React Query implementation anyway

**Comment by TKDickson** [comment]
- [ ] When a Veteran does an action that affects the feature values displayed on the home screen (read a secure message, cancel an appointment, fill a prescription that is available for refill), then the feature value's state is captured and reflected on the home screen
- [ ] User does NOT take action that would change the home screen - we re-call all of the home screen APIs every 5 minutes (upon navigation to the home screen) and all sections show the 're-loading' UI state
- [ ] User DOES take action that changes the home screen, and it's been longer than 5 minutes - we re-call all APIs (like immediately above)
- [ ] Maintain scroll position on backtrack to Homescreen from any of the links
- [ ] During dev/QA, should test that the loading behavior previously on the home screen when returning from "Contact VA" does not demonstrate the jumpiness in this [video] (https://drive.google.com/file/d/1fe415bCgN77R-LqtPtZfGyzse04t7RbW/view?usp=sharing).

**Comment by TKDickson** [comment]
Actions, in-app:
- [ ] Cancel an appointment
- [ ] Read a standalone message
- [ ] Read the latest message, in a thread with other unread messages
- [ ] Read a middle message, in a thread with newer and older unread messages
- [ ] Fill an RX

Outside of app actions (after logging into app/during the 'not refreshed' 5 minute window):
- [ ] Cancel an appointment (website)
- [ ] Read a standalone message (website)
- [ ] Read the latest message, in a thread with other unread messages (website)
- [ ] Read a middle message, in a thread with newer and older unread messages (website)
- [ ] Fill an RX (website)
- [ ] Schedule an appointment (website/if I can, the setup here is sometimes busted)
- [ ] Send a new message (clinician)

**Comment by TKDickson** [comment]
This is impacted by #8056 (which was not fixed with the SM RQ migration, unfortunately). Pinging a few folks to figure out who would fix/when it might get fixed.

The "thread-related" message tests above all therefore fail.

**Comment by TKDickson** [comment]
We currently don't reset the refresh timer at logout.

So - log in as veteran A. Log out. Log in as veteran B and see veteran A's home screen activity.

Sending back for changes.

**Comment by theodur** [comment]
@TKDickson I fixed the issue where the cache wasn't being reset on logout. Should be good now to take another look

**Comment by TKDickson** [comment]
We cache appointments and claims/appeals for 30 minutes on the mobile backend; we also cache messages for 20 minutes. Prescriptions is not cached.

This simplifies/removes a number of cases I had previously considered (and I will go update [the relevant comment](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8183#issuecomment-2093398450) accordingly). It also is information we didn't know when we made the decision to "go get new information and show the loading UI" every 5 minutes for home screen stuff, and so we need to figure out if that information makes us want to do something different. Brought up at an 11th minute today - unfortunately key folks weren't there - so @dumathane is going to make sure to bring it up with @htcollier even if I'm out for parts/all of this week.

**Comment by TKDickson** [comment]
Blocked from testing prescriptions because karsyn is out of refillable meds (MHV team is working on it; some other team appears to be using karsyn's credentials - maybe even for automated testing - because 30 refills is typically disappearing in ~1-2 weeks, at this point)

**Comment by TKDickson** [comment]
Karsyn has meds now, confirmed that Rx counts change (category and home) and this ticket is closeable.
